src/Pages/Withdrawals.razor

```diff
@@ -521,8 +524,13 @@
 
     private async Task GetData()
     {
-        if (LoggedUser?.Id != null)
+        if (LoggedUser?.Id != null && !_isFinanceManager)
+        {
             _userPendingRequests = await WalletWithdrawalRequestRepository.GetUnsignedPendingRequestsByUser(LoggedUser.Id);
+        }else 
+        {
+            _userPendingRequests = await WalletWithdrawalRequestRepository.GetAllUnsignedPendingRequests(); 
+        }
```

Comment:
you don't need the first function anymore, because now it doesn't matter who the user is when fetching, only when displaying the button. Remember that the table above doesn't show if the user is not a finance manager. So you can simplify all this by


```
if (LoggedUser?.Id != null && _isFinanceManager)
{
   _userPendingRequests = await WalletWithdrawalRequestRepository.GetAllUnsignedPendingRequests(); 
}
```

Check if `GetUnsignedPendingRequestsByUser` is used elsewhere, otherwise you can remove it entirely from the code

Reply:
Yes, it make sense.